src/libaktualizr/primary/download_nonostree_test.cc

```diff
@@ -0,0 +1,111 @@
+#include <gtest/gtest.h>
+
+#include <boost/process.hpp>
+
+#include "uptane_test_common.h"
+
+#include "config/config.h"
+#include "logging/logging.h"
+#include "package_manager/ostreemanager.h"
+#include "primary/aktualizr.h"
+#include "storage/sqlstorage.h"
+#include "test_utils.h"
+
+static std::string server = "http://127.0.0.1:";
+static std::string treehub_server = "http://127.0.0.1:";
+static boost::filesystem::path sysroot;
+
+static std::string new_rev;
+
+/*
+ * Reject non-OSTree binaries.
+ */
+TEST(Aktualizr, DownloadNonOstreeBin) {
+  TemporaryDirectory temp_dir;
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, server);
+  conf.pacman.type = PackageManager::kOstree;
+  conf.pacman.sysroot = sysroot.string();
+  conf.pacman.ostree_server = treehub_server;
+  conf.pacman.os = "dummy-os";
+  conf.provision.device_id = "device_id";
+  conf.provision.ecu_registration_endpoint = server + "/director/ecus";
+  conf.tls.server = server;
+
+  {
+    std::shared_ptr<INvStorage> storage = INvStorage::newStorage(conf.storage);
+    OstreeManager ostree(conf.pacman, storage, nullptr, nullptr);
+
+    Aktualizr aktualizr(conf);
+
+    aktualizr.Initialize();
+
+    result::UpdateCheck update_result = aktualizr.CheckUpdates().get();
+    ASSERT_EQ(update_result.status, result::UpdateStatus::kError);
+  }
```

Comment:
Alright, I think I figured it out. The problem is that `CheckUpdates` calls `PutManifest`, which calls `getManifest`, which calls `getCurrent`, which doesn't work with a fake sysroot. Solution: work at a lower level, since we don't need to do any of that. I suggest replacing the section in braces here with this code:

```
    std::shared_ptr<INvStorage> storage = INvStorage::newStorage(conf.storage);
    auto uptane_client = SotaUptaneClient::newDefaultClient(conf, storage);
    uptane_client->initialize();
    EXPECT_FALSE(uptane_client->uptaneIteration());
    EXPECT_STREQ(uptane_client->getLastException().what(),
                 "The target had a non-OSTree package that can not be installed on an OSTree system.");
```